src/LandBoundaries.cpp

```diff
@@ -43,18 +43,21 @@ namespace meshkernel
         m_mesh(mesh), 
         m_polygons(polygons)
     {
-        ResizeVectorIfNeededWithMinimumSize(m_numAllocatedNodes, m_nodes, m_allocationSize, { doubleMissingValue,doubleMissingValue });
-        m_numAllocatedNodes = int(m_nodes.size());
         m_numNode = 0;
-        m_polygonNodesCache.resize(maximumNumberOfNodesPerFace);
+        m_numAllocatedNodes = 0;
+        if (!landBoundary.empty())
+        {
+            ResizeVectorIfNeededWithMinimumSize(int(landBoundary.size()), m_nodes, m_allocationSize, { doubleMissingValue,doubleMissingValue });
+            m_numAllocatedNodes = int(m_nodes.size());
 
-        ResizeVectorIfNeededWithMinimumSize(m_numNode + int(landBoundary.size()), m_nodes, m_allocationSize, { doubleMissingValue,doubleMissingValue });
-        m_numAllocatedNodes = int(m_nodes.size());
+            m_numNode = 0;

```

Comment:
Why do you set `m_numNode` to zero twice?

Reply:
See my second commit (a simplification). in my first commit m_numNode needs to be assigned to a nongarbage value in case the landboundary line is empty